app/src/main/java/com/alexchan/wallpaper/ui/wallpaper/dashboard/dashboard/DashboardFragment.kt

```diff
@@ -0,0 +1,261 @@
+
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import android.content.res.Configuration
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.StaggeredGridLayoutManager
+import com.alexchan.wallpaper.R
+import com.alexchan.wallpaper.adapter.photoBinding.PhotoGridAdapter
+import com.alexchan.wallpaper.databinding.FragmentDashboardBinding
+import com.alexchan.wallpaper.ui.mainActivity.MainActivity
+import com.alexchan.wallpaper.ui.search.searchActivity.SearchActivity
+import com.alexchan.wallpaper.ui.wallpaper.wallpaper.WallpaperFragmentDirections
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.fragment_dashboard.*
+
+
+class DashboardFragment : Fragment() {
+
+    private val dashboardViewModel: DashboardViewModel by lazy {
+        ViewModelProvider(this).get(DashboardViewModel::class.java)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val binding = FragmentDashboardBinding.inflate(inflater)
+
+        // Allows Data Binding to Observe LiveData with the lifecycle of this Fragment
+        binding.lifecycleOwner = this
+
+        // Giving the binding access to the DashboardViewModel
+        binding.dashboardViewModel = dashboardViewModel
+
+        binding.photosGrid.adapter = PhotoGridAdapter(PhotoGridAdapter.OnClickListener {
+            dashboardViewModel.displayUserCollection(it)
+        })
+        dashboardViewModel.navigateToUserCollection.observe(viewLifecycleOwner, Observer {
+            if (null != it) {
+                this.findNavController().navigate(
+                    WallpaperFragmentDirections.actionShowDetail(it))
+                dashboardViewModel.displayUserCollectionComplete()
+            }
+        })
+
+        // Show Top Toolbar
+        requireActivity().topToolbar.visibility = View.VISIBLE
+
+        // Set TopToolbar OnMenuItemClickListener
+        requireActivity().topToolbar.setOnMenuItemClickListener{item: MenuItem? -> onMenuItemClick(item)}
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // Reset Search Query
+        MainActivity.searchQuery = ""
```

Comment:
Suggestion: Make the search bar just for display, when clicked navigation to another screen and handle all the search there.